File: backend/src/main/java/dev/coldhands/pair/stairs/backend/infrastructure/persistance/repository/TeamRepository.java
package dev.coldhands.pair.stairs.backend.infrastructure.persistance.repository;

import dev.coldhands.pair.stairs.backend.infrastructure.persistance.entity.TeamEntity;
import org.springframework.data.jpa.repository.JpaRepository;

public interface TeamRepository extends JpaRepository<TeamEntity, Long> {

    TeamEntity findBySlug(String slug);
}

File: backend/src/main/kotlin/dev/coldhands/pair/stairs/backend/infrastructure/persistance/repository/TeamRepository.kt
package dev.coldhands.pair.stairs.backend.infrastructure.persistance.repository

import dev.coldhands.pair.stairs.backend.infrastructure.persistance.entity.TeamEntity
import org.springframework.data.jpa.repository.JpaRepository

interface TeamRepository : JpaRepository<TeamEntity, Long> {
    fun findBySlug(slug: String?): TeamEntity?
}